erpnext/stock/stock_ledger.py

```diff
@@ -381,11 +380,9 @@ def __init__(
 		self.initialize_previous_data(self.args)
 		self.build()
 
-	def get_precision(self):
-		company_base_currency = frappe.get_cached_value("Company", self.company, "default_currency")
-		self.precision = get_field_precision(
-			frappe.get_meta("Stock Ledger Entry").get_field("stock_value"), currency=company_base_currency
-		)
+	def set_precision(self):
+		self.flt_precision = cint(frappe.db.get_default("float_precision")) or 2
+		self.currency_precision = cint(frappe.db.get_default("currency_precision")) or 2
```

Comment:
This is often not set and derived from number format 😐 

https://github.com/frappe/frappe/blob/a59e5d7aacf2ddcea8885591c191e9c93f7984d0/frappe/model/meta.py#L730-L733

Reply:
@ankush but the fallback is in the case that currency precision is not set and somehow it doesnt get anything from number format also. Most cases its not likely to use the fallback `or 2`

Reply:
No, I meant on most sites currency precision is not defined. It's derived from the number format selected by the user. 

Fallback of 2 is fine.

Reply:
https://github.com/frappe/frappe/blob/develop/frappe/core/doctype/system_settings/system_settings.json#L162

`currency_precision` has no defaults and is not mandatory also. So on most sites this field wont have any value.

Reply:
@marination IMO we should not change this rounding behaviour of currency now, it will cause different rounding when new code is applied. 


This is what I am suggesting:

```diff
diff --git a/erpnext/stock/stock_ledger.py b/erpnext/stock/stock_ledger.py
index 3afa899e05..66086dcdbb 100644
--- a/erpnext/stock/stock_ledger.py
+++ b/erpnext/stock/stock_ledger.py
@@ -382,7 +382,6 @@ class update_entries_after(object):
 
 	def set_precision(self):
 		self.flt_precision = cint(frappe.db.get_default("float_precision")) or 2
-		self.currency_precision = cint(frappe.db.get_default("currency_precision")) or 2
 
 	def initialize_previous_data(self, args):
 		"""
@@ -578,7 +577,7 @@ class update_entries_after(object):
 					self.update_queue_values(sle)
 
 		# rounding as per precision
-		self.wh_data.stock_value = flt(self.wh_data.stock_value, self.currency_precision)
+		self.wh_data.stock_value = flt(self.wh_data.stock_value, self.precision)
 		if not self.wh_data.qty_after_transaction:
 			self.wh_data.stock_value = 0.0
 		stock_value_difference = self.wh_data.stock_value - self.wh_data.prev_stock_value
```

Reply:
will leave `stock_value` as is

Reply:
@ankush `stock_value` now uses `get_field_precision` (consistent behaviour as before).
Rest, use precision only for error message data (consistent with what users can actually input to restore data as per precision)